ledfx/virtuals.py

```diff
@@ -353,7 +354,14 @@ def active_effect(self):
         return self._active_effect
 
     def thread_function(self):
+        self.passed = 0.0
+        self.sleep_time = 0.0
+        self.postsleep = 0.0
+        self.presleep = 0.0
+
         while True:
+            self.start = timeit.default_timer()
+            _LOGGER.warning(f"presleep {self.presleep:.04f} postsleep:{self.postsleep:.04f} sleep for: {self.sleep_time:.04f} passed: {self.passed}")
```

Comment:
comment this line out to stop the per frame spam, and just watch changes on switching sources